### ISSUE TITLE
Optionally add task URL to changeset comment

### ISSUE THREAD

**Issue by mvexel** [osmlab/maproulette3#178]
so that it reads this by default, user can always override.

<img width="1129" alt="screenshot 2018-03-13 19 17 48" src="https://user-images.githubusercontent.com/120452/37378088-58a44348-26f3-11e8-9520-f87d2df2f6a9.png">

**Comment by nrotstan** [comment]
Is this a duplicate of #5 with an alternative approach, or is it different?

One minor hurdle to the pre-fill approach suggested in this issue is that we're not creating the challenge on the server until the end of the workflow, so we don't currently have the challenge id available at this point. But we could change that, or use some kind of placeholder to represent the link instead (`{{challengeLink}}` or something) that gets replaced with the proper URL once the challenge id is known. The checkbox approach suggested in #5 sidesteps this issue, but also would reduce the user's flexibility in determining where that link appears in the changeset comment.

**Comment by mvexel** [comment]
It's more of a superset of #5. I'll close that one in favor of this. Thinking about this some more, perhaps the way to go would be as follows:

* Do not pre-fill the text box
* Add two checkboxes / switches:
  * "Add `#maproulette` hashtag (checked by default) -- would add `#maproulette` to the changeset comment
  * "Add link to task" (checked by default as well) -- would add a link to the current task to the changeset comment

I think we can just prescribe the order of these, I doubt people will care all that much. If there is a need, we may add a checkbox in the user profile settings to allow users to ignore maproulette-generated changeset comments.

**Comment by nrotstan** [comment]
PR #292 addresses the first half of this issue (#maproulette hashtag) and will be in the beta.5 release.

**Comment by mvexel** [comment]
Right. #290 duplicates some of this..

**Comment by mvexel** [comment]
So that leaves the second part. I would still like to see a link to the task, optionally. Perhaps we can use #225 to not clutter the comments too much. Not the most urgent thing but definitely nice to have. Will rename issue to reflect smaller scope.

**Comment by mvexel** [comment]
Revisiting this -- I am keeping this one alive. I think we should use the short URL https://mpr.lt/<task_id>. For mappers to verify the work it would be good to have a direct link to the original task.

**Comment by nrotstan** [comment]
Agreed @mvexel . I think what we're mostly waiting on here is deciding what the landing page should look like when someone follows that link. I don't think we can assume that someone clicking a link from an OSM changeset is necessarily familiar with MapRoulette -- or has even heard of it -- and just dropping them straight into the task-completion page might not be ideal (I think we may also require them to be logged in to get to that page). I think it would make sense to have a dedicated landing page that introduces MapRoulette, describes the challenge, and then also presents the task in question (and obviously wouldn't require the user to be logged in to MR).

Also, once we start including task links in OSM changesets, we're essentially committing ourselves to preserving that task data forever in some form so that the links will work long after the challenge is completed. I don't see that as a problem, but it's something to keep in mind.

**Comment by mvexel** [comment]
I had thought of the second point but not the first. A landing page does make sense. Let's think about what that should contain. I think you're mentioning some of the important elements:

* The basic task info (map, instruction, any history)
* The parent Challenge info (description, history, static map perhaps with convex hull or bbox depicting task locations?)
* Link to contact the challenge owner
* An OSMCha or similar link that would show other changesets related to this challenge (this would require the shortlink to be `https://mpr.lt/<challenge_id>/<task_id>`)
* One or two paragraphs about what MapRoulette is (with link to a video?)

**Comment by mvexel** [comment]
Bumping this -- we have the PR still pending and I think it's good to have the feedback loop for people stumbling on OSM changesets that have a #maproulette reference.

**Comment by mvexel** [comment]
Reopening this -- right now we're including a CHALLENGE URL, whereas it is more appropriate to link a changeset to the specific TASK it is linked to.

**Comment by mvexel** [comment]
Thinking about this more, it would be nice if we had some sort of info-box / modal that would be shown if the referer is a changeset URL. The people clicking on a MR link from an OSM changeset are a) likely to be unfamiliar with the MR interface, or with MR altogether, and b) likely to have some sort of concern or complaint that led them to click on the link.

So, without having the exact wording just yet, this message should contain:
* A (link to) a basic explanation of what MapRoulette is
* A quick explanation of what the relation is between the changeset and this task
* Instructions on how to get in touch with the challenge owner (or perhaps even a direct link to do this)

**Comment by wolfgang-8** [comment]
Please also add a link to the MR task page. This would be handy for MR users who want to add comments afterwards, change MR tags or just check task status after a task is closed.

**Comment by 1ec5** [comment]
Without this task URL, individual participants in MapRoulette challenges can get called out for things that are already explained well in the challenge description, such as in [this changeset discussion](https://www.openstreetmap.org/changeset/122094142).

**Comment by mvexel** [comment]
I created #1873 to capture my comment above.

The feature is otherwise implemented. If anything additional is desired, please open a new issue. Thanks!
